service/src/main/java/bio/terra/workspace/service/policy/flight/ValidateWorkspaceAgainstPolicyStep.java

```diff
@@ -0,0 +1,88 @@
+package bio.terra.workspace.service.policy.flight;
+
+import bio.terra.policy.model.TpsPaoGetResult;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.policy.TpsApiDispatch;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import bio.terra.workspace.service.resource.exception.PolicyConflictException;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import java.util.HashSet;
+import java.util.List;
+import java.util.UUID;
+
+public class ValidateWorkspaceAgainstPolicyStep implements Step {
+  private final UUID workspaceId;
+  private final CloudPlatform cloudPlatform;
+  private final String destinationLocation;
+  private final AuthenticatedUserRequest userRequest;
+  private final ResourceDao resourceDao;
+  private final TpsApiDispatch tpsApiDispatch;
+
+  public ValidateWorkspaceAgainstPolicyStep(
+      UUID workspaceId,
+      CloudPlatform cloudPlatform,
+      String destinationLocation,
+      AuthenticatedUserRequest userRequest,
+      ResourceDao resourceDao,
+      TpsApiDispatch tpsApiDispatch) {
+    this.workspaceId = workspaceId;
+    this.cloudPlatform = cloudPlatform;
+    this.userRequest = userRequest;
+    this.tpsApiDispatch = tpsApiDispatch;
+    this.resourceDao = resourceDao;
+    this.destinationLocation = destinationLocation;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    final TpsPaoGetResult effectivePolicies =
+        flightContext
+            .getWorkingMap()
+            .get(WorkspaceFlightMapKeys.EFFECTIVE_POLICIES, TpsPaoGetResult.class);
+
+    if (cloudPlatform != CloudPlatform.GCP) {
+      // We can only validate GCP right now. We'll need to add other platforms
```

Comment:
i thought tps knows azure regions

Reply:
It's built to support it, but it doesn't do much with it right now. We took Azure out of the Ontology and aren't validating regions on resource create right now.